provision/aws/rosa_delete_cluster.sh

```diff
@@ -23,22 +23,23 @@ if [ -z "$REGION" ]; then echo "Variable REGION needs to be set."; exit 1; fi
 ./rds/aurora_delete_peering_connection.sh || true
 ./rosa_efs_delete.sh || true
 
+
+pids=()
 #Creating Logs directory for each cluster
 LOG_DIR="${SCRIPT_DIR}/logs/${CLUSTER_NAME}"
 mkdir -p ${LOG_DIR}
 cd ${LOG_DIR}
 echo "Starting to watch cluster uninstall logs."
-rosa logs uninstall -c ${CLUSTER_NAME} --watch --debug 2>&1 | tee "$(custom_date)_delete-cluster.log" &
-LOGS_SAVING_PROC_ID=$!
+timeout -k 30m 30m bash -c "rosa logs uninstall -c ${CLUSTER_NAME} --watch --debug &> $(custom_date)_delete-cluster.log" &
+pids+=($!)
 
 cd ${SCRIPT_DIR}/../opentofu/modules/rosa/hcp
 WORKSPACE=${CLUSTER_NAME}-${REGION}
-./../../../destroy.sh ${WORKSPACE} &
-DESTROY_PROC_ID=$!
+timeout -k 30m 30m bash -c "./../../../destroy.sh ${WORKSPACE}" &
+pids+=($!)
 
-timeout 30m bash -c 'wait $DESTROY_PROC_ID $LOGS_SAVING_PROC_ID'
-if [[ $? -eq 124 ]]; then
-    echo "Timeout occurred after 30 minutes."
-else
+if wait "${pids[@]}"; then
```

Comment:
@ryanemerson I have tried to run a simple script on my local using timeout / and `if wait`, as it is done here. So `wait "${pids[@]}"` returns false only when both processes exited with timeout. If one of them finished with timeout and other one normally exited, then the `wait` will return true and will print that everything is OK.

Reply:
Good spot @andyuk1986. I've updated it so that we only wait on the destroy process, as once that's completed the `rosa logs` command will terminate due to a 404 anyway.

Reply:
@ryanemerson thank a lot. Now it looks good.